File: src/modules/blog/domain/__tests__/entities/post.test.ts
import { PostMother } from '../mothers/post.mother';
import { AuthorMother } from '../mothers/author.vo.mother';
import { Post } from '../../entities/post';
import { Author } from '../../valueObjects/author.vo';

describe('Post tests', () => {
  it('should create a post', () => {
    const post = PostMother.createRandom();
    expect(post).toBeDefined();
  });

  it('should create a Post with data', () => {
    const post = PostMother.createWithData(
      'id-1',
      'title-1',
      'slug-1',
      'content-1',
      new Date(),
      new Date(),
      'image-1',
      AuthorMother.createWithData('Rodia'),
    );
    expect(post).toBeDefined();
    expect(post.id).toBe('id-1');
    expect(post.title).toBe('title-1');
    expect(post.content).toBe('content-1');
    expect(post.createdAt).toBeInstanceOf(Date);
    expect(post.updatedAt).toBeInstanceOf(Date);
    expect(post.image).toBe('image-1');
    expect(post.author.name).toBe('Rodia');
    expect(post.section).toBe('blog');
  });

  it('should create an empty post', () => {
    const post = PostMother.createEmpty();
    expect(post).toBeDefined();
    expect(post.id).toBe('');
    expect(post.title).toBe('');
    expect(post.content).toBe('');
    expect(post.createdAt).toBeInstanceOf(Date);
    expect(post.updatedAt).toBeInstanceOf(Date);
    expect(post.image).toBe('');
    expect(post.author.name).toBe('');
    expect(post.section).toBe('');
  });

  it('should create a random post', () => {
    const post = PostMother.createRandom();
    expect(post).toBeDefined();
  });

  describe('calculateReadingTime', () => {
    it('should return 1 minute for empty content', () => {
      const post = Post.create(
        'id',
        'title',
        'slug',
        '',
        new Date(),
        new Date(),
        'image',
        Author.empty(),
      );
      expect(post.calculateReadingTime()).toBe(1);
    });

    it('should return 1 minute for short content', () => {
      const post = Post.create(
        'id',
        'title',
        'slug',
        'word '.repeat(100),
        new Date(),
        new Date(),
        'image',
        Author.empty(),
      );
      expect(post.calculateReadingTime()).toBe(1);
    });

    it('should calculate correct minutes for longer content', () => {
      const post = Post.create(
        'id',
        'title',
        'slug',
        'word '.repeat(400),
        new Date(),
        new Date(),
        'image',
        Author.empty(),
      );
      // 400 words / 200 wpm = 2 minutes
      expect(post.calculateReadingTime()).toBe(2);
    });
  });
});

File: test/domain/entities/post_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:staretz/domain/entities/post.dart';
import 'package:staretz/domain/value_objects/title.dart';
import 'package:staretz/domain/value_objects/author.dart';
import 'package:staretz/domain/value_objects/content.dart';
import 'package:staretz/domain/value_objects/tag.dart';
import 'package:staretz/domain/value_objects/slug.dart';

import '../../utils/post_mother.dart';

void main() {
  group('Post Entity', () {
    test('should create a valid Post from Post.create', () {
      final titleStr = 'My Test Title';

      final post = Post.create(
        id: '123',
        title: Title(titleStr),
        author: const Author('John Doe'),
        createdAt: DateTime(2025, 1, 1),
        content: const Content('Content body'),
        tags: const [Tag('test')],
        slug: const Slug('my-test-title'),
        images: const [],
      );

      expect(post.id, '123');
      expect(post.title.value, titleStr);
      expect(post.author.value, 'John Doe');
      expect(post.content.value, 'Content body');
    });

    test('should create an empty post using Post.empty', () {
      final post = Post.empty();

      expect(post.id, isEmpty);
      expect(post.title.value, isEmpty);
      expect(post.author.value, isEmpty);
      expect(post.tags, isEmpty);
      expect(post.images, isEmpty);
    });

    test('should create a valid post using PostMother', () {
      final post = PostMother.create(title: const Title('Mother Title'));

      expect(post.id, isNotEmpty);
      expect(post.title.value, 'Mother Title');
    });
  });
}
